### ISSUE TITLE
TypeError: optimism.dep is not a function. (In 'optimism.dep()', 'optimism.dep' is undefined)

### ISSUE THREAD

**Issue by dhavaljardosh** [apollographql/apollo-client#7405]
I am migrating from `react-apollo` to `@apollo/client`, have tried a lot of things but was unable to fix this issue.

![Error](https://i.ibb.co/8MZmpc3/Screen-Shot-2020-12-02-at-12-39-07-AM.png)

**package.json**
```
"dependencies": {
    "@apollo/client": "^3.2.9",
    "@apollo/link-context": "^2.0.0-beta.3",
    "@react-native-community/async-storage": "~1.12.0",
    "@react-native-community/datetimepicker": "3.0.0",
    "@react-native-community/viewpager": "^4.1.6",
    "@react-navigation/drawer": "^5.11.4",
    "crypto": "^1.0.1",
    "crypto-js": "^4.0.0",
    "expo": "^39.0.3",
    "expo-facebook": "~9.0.0",
    "expo-font": "~8.3.0",
    "expo-google-app-auth": "^8.0.0",
    "expo-image-picker": "~9.1.0",
    "expo-notifications": "~0.7.2",
    "graphql": "^14.4.2",
    "moment": "^2.24.0",
    "native-base": "^2.13.8",
    "react": "16.13.1",
    "react-dom": "16.13.1",
```
    "react-native": "https://github.com/expo/react-native/archive/sdk-39.0.0.tar.gz",
    "react-native-aws3": "^0.0.9",
    "react-native-gesture-handler": "~1.7.0",
    "react-native-google-places-autocomplete": "^1.4.1",
    "react-native-maps": "0.27.1",
    "react-native-material-ripple": "^0.9.1",
    "react-native-modal": "^11.5.6",
    "react-native-modal-datetime-picker": "^9.0.0",
    "react-native-multiple-select": "^0.5.3",
    "react-native-paper": "^4.4.1",
    "react-native-picker-select": "^6.3.3",
    "react-native-screens": "~2.10.1",
    "react-native-status-bar-height": "^2.4.0",
    "react-native-tab-view": "^2.15.2",
    "react-native-testing-library": "^6.0.0",
    "react-native-vector-icons": "^6.6.0",
    "react-native-web": "~0.13.7",
    "react-navigation": "^4.4.2",
    "react-navigation-drawer": "^1.4.0",
    "react-navigation-stack": "^1.7.3",
    "react-navigation-tabs": "^1.2.0"
  },

**Comment by benjamn** [comment]
@dhavaljardosh What `version` do you see in `node_modules/optimism/package.json`? It should be 0.13.1, since [that's what `@apollo/client` requires](https://github.com/apollographql/apollo-client/blob/0ef205b049e518adb7d3ce10f4bc2e7ee556ea82/package.json#L81).

**Comment by dhavaljardosh** [comment]
@benjamn yes I did verify that `optimism` has 0.13.1. Btw was stuck on this for 2 days, did cache cleaning, package installation and re-installation and everything and nothing worked. 

And miraculously right now it worked. And its without any cache or npm installation.
Let me see if this issue pops up again(will monitor it for a day), if it doesn't will close the issue.

**Comment by noudadrichem** [comment]
Got the same error while building a NextJS project with `node:14-alpine` image.
This error occured after updating `@apollo/client` from `3.1.4` to `3.3.3`

**Comment by dhavaljardosh** [comment]
A few steps that I kept on doing here was:
- deleting the **node_modules** and **package-lock.json** and do `npm install` or `yarn install`
- removing cache before running
- and the most common computer fix, restarting the system

I was struggling with this for around 2 days with no output but was doing the above steps over and over and at one point it's working fine.

And as @benjamn suggested `optimism` should have the version `1.3.0` which I did all the time but my guess is that wasn't the problem.

Closing it as couldn't find any concrete solution but it started working fine for me.
